api/src/repositories/user-repository.ts

```diff
@@ -407,4 +408,56 @@ export class UserRepository extends BaseRepository {
       ]);
     }
   }
+
+  /**
+   * Update a system user's information.
+   *
+   * Note: Attempts to find a system user record using the `user_guid` first, and if no record is found, attempts
+   * using the `user_identifier` + `user_identity_source`. If a record is found, the record is updated and the
+   * system user id is returned. If no user is found, does nothing and returns null.
+   *
+   * @param {({
```

Comment:
it would be nice to see this in an interface that can be used in the service as well

Reply:
Yeah I can make one, was just being lazy